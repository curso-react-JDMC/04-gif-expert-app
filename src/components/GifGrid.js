import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs';
import GifGridItem from './GifGridItem';
import PropTypes from 'prop-types';

const GifGrid = ({ category }) => {
  const { data: images, loading } = useFetchGifs(category);
  //console.log(loading);

  // getGifs();
  return (
    <>
      <h3 className="animate_animated animate_fadeIn">{category}</h3>
      {loading && <p>Loading</p>}
      <div className="card-grid">
        <ol>
          {
            images.map((img) => (
              // <li key={ id }>{ title }</li>
              <GifGridItem key={img.id} {...img} />
            ))
          }
        </ol>
      </div>
    </>
  )
}

GifGrid.propTypes = {
  category: PropTypes.string.isRequired
}
export default GifGrid;
