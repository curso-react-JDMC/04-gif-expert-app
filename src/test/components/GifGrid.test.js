import React from 'react';
import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import GifGrid from "../../components/GifGrid";
import { useFetchGifs } from '../../hooks/useFetchGifs';
jest.mock('../../hooks/useFetchGifs');

describe('pruebas en GifGrid', () => {
  const category = 'Harry potter';

  test('debe mostrarse correctamente', () => {

    useFetchGifs.mockReturnValue({
      data: [],
      loading: true
    });
    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper).toMatchSnapshot();
  })

  test('debe mostrar items cuando se cargan imÃ¡genes useFetchGifs', () => {
    const gifs = [{
      id: 'ABC',
      url: 'http://google.com',
      title: 'ABC title',
    }]
    useFetchGifs.mockReturnValue({
      data: gifs,
      loading: false
    });

    const wrapper = shallow(<GifGrid category={category} />)
    expect(wrapper.find('p').exists()).toBe(false);
    expect(wrapper.find('GifGridItem').length).toBe(gifs.length)
  })

})
