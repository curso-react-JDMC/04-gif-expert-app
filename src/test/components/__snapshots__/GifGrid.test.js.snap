// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pruebas en GifGrid debe mostrarse correctamente 1`] = `
<Fragment>
  <h3
    className="animate_animated animate_fadeIn"
  >
    Harry potter
  </h3>
  <p>
    Loading
  </p>
  <div
    className="card-grid"
  >
    <ol />
  </div>
</Fragment>
`;
